File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int solution(int[] queue1, int[] queue2) {
        int answer = 0;
        var q1 = new Queue<int>(queue1);
        var q2 = new Queue<int>(queue2);
        
        long sum1 = 0, sum2 = 0;
        foreach (var i in queue1) sum1 += i;
        foreach (var i in queue2) sum2 += i;
        
        long target = (sum1 + sum2) / 2;
        int limit = (queue1.Length + queue2.Length) * 2;
        
        while(answer <= limit && sum1 != target)
        {
            if(sum1 > target)
            {
                if(q1.Count == 0) return -1;
                int v = q1.Dequeue();
                sum1 -= v;
                q2.Enqueue(v);
            }
            else
            {
                if(q2.Count == 0) return -1;
                int v = q2.Dequeue();
                sum1 += v;
                q1.Enqueue(v);
            }
            answer++;
        }
        return sum1 == target ? answer : -1;
    }
}

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

int solution(vector<int> queue1, vector<int> queue2) {
    queue<int> q1, q2;
    long long sum1 = 0, sum2 = 0;
    int answer = 0;
    
    for(int x : queue1)
    {
        q1.push(x);
        sum1 += x;
    }
    for(int x : queue2)
    {
        q2.push(x);
        sum2 += x;
    }
    
    long long target = (sum1 + sum2) / 2;
    int limit = queue1.size() * 3;
    
    while(answer <= limit)
    {
        if(sum1 == target) return answer;
        
        if(sum1 > target)
        {
            int x = q1.front();
            q1.pop();
            q2.push(x);
            sum1 -= x;
        }
        else
        {
            int x = q2.front();
            q2.pop();
            q1.push(x);
            sum1 += x;
        }
        answer++;
    }
    
    return -1;
}